server/cromwell.py

```diff
@@ -0,0 +1,247 @@
+                    user=email,
+                    access_level=access_level,
+                    repo=repo,
+                    commit=commit,
+                    script=wf,
+                    description=params['description'],
+                    output=output_dir,
+                    driver_image=DRIVER_IMAGE,
+                    cwd=cwd,
+                    mode='cromwell',
+                )
+            )
+
+            # Publish the metadata to Pub/Sub. Wait for the result before running the batch.
+            publisher.publish(PUBSUB_TOPIC, metadata.encode('utf-8')).result()
+
+            user_name = email.split('@')[0]
+            batch_name = f'{user_name} {repo}:{commit}/cromwell/{wf}'
+
+            batch = hb.Batch(backend=backend, name=batch_name)
+
+            job = batch.new_job(name='driver')
+            job = prepare_git_job(
+                job=job,
+                dataset=dataset,
+                access_level=access_level,
+                repo=repo,
+                commit=commit,
+                metadata_str=metadata,
+                output_dir=output_dir,
+                print_all_statements=False,
+            )
+            job.env('OUTPUT', output_dir)
+            # we'll need to put the commands like this, because
+            # the job.command is individually scoped, ie, the cd
+            # doesn't carry across the different job.commands
+            commands = []
+            if cwd:
+                commands.append(f'cd {quote(cwd)}')
+
+            deps_path = None
+            if libs:
+                deps_path = 'tools.zip'
+                commands.append(
+                    'zip -r tools.zip ' + ' '.join(quote(s + '/') for s in libs)
+                )
+
+            cromwell_post_url = CROMWELL_URL + '/api/workflows/v1'
+            workflow_options = {
+                'user_service_account_json': service_account_json,
+                'google_compute_service_account': service_account_email,
+                'google_project': project,
+                'jes_gcs_root': intermediate_dir,
+                'final_workflow_outputs_dir': output_dir,
+            }
+
+            if input_dict:
+                tmp_input_json_path = '/tmp/inputs.json'
+                commands.append(
+                    f"echo '{json.dumps(input_dict)}' > {tmp_input_json_path}"
+                )
+                input_jsons.append(tmp_input_json_path)
+
+            inputs_cli = []
+            for idx, value in enumerate(input_jsons):
+                key = 'workflowInputs'
+                if idx > 0:
+                    key += f'_{idx + 1}'
+
+                inputs_cli.append(f'-F "{key}=@{value}"')
+
+            nl = '\n'
+            job.command(
+                f"""
+{nl.join(commands)}
+echo '{json.dumps(workflow_options)}' > workflow-options.json
+access_token="$(gcloud auth print-identity-token --audiences="717631777761-ec4u8pffntsekut9kef58hts126v7usl.apps.googleusercontent.com")"
```

Comment:
Those nested quotes look brittle.
```suggestion
access_token=$(gcloud auth print-identity-token --audiences=717631777761-ec4u8pffntsekut9kef58hts126v7usl.apps.googleusercontent.com)
```